File: Interview Problems/Parking Lot/Java Code/repository/FloorRepository.java
package repository;

import domain.Floor;

import java.util.HashMap;
import java.util.Map;
import java.util.Optional;

public class FloorRepository {

    // Keyed by floorNumber for fast lookup
    private final Map<Integer, Floor> store = new HashMap<>();

    public void save(Floor floor) {
        store.put(floor.getFloorNumber(), floor);
        System.out.println("[REPO] Floor saved: " + floor.getFloorNumber());
    }

    public Optional<Floor> findByFloorNumber(int floorNumber) {
        return Optional.ofNullable(store.get(floorNumber));
    }

    public boolean existsByFloorNumber(int floorNumber) {
        return store.containsKey(floorNumber);
    }

    public Map<Integer, Floor> findAll() {
        return store;
    }
}

File: Interview Problems/Parking Lot/Python Code/repository/floor_repository.py
from typing import Dict, Optional
from domain.floor import Floor

class FloorRepository:
    def __init__(self):
        self._store: Dict[int, Floor] = {}

    def save(self, floor: Floor):
        self._store[floor.floor_number] = floor
        print(f"[REPO] Floor saved: {floor.floor_number}")

    def find_by_floor_number(self, floor_number: int) -> Optional[Floor]:
        return self._store.get(floor_number)

    def exists_by_floor_number(self, floor_number: int) -> bool:
        return floor_number in self._store

    def find_all(self) -> Dict[int, Floor]:
        return self._store
